galasa-managers-parent/galasa-managers-visualisation-parent/dev.galasa.elasticlog.manager/src/main/java/dev/galasa/elasticlog/internal/ElasticLogManagerImpl.java

```diff
@@ -0,0 +1,227 @@
+package dev.galasa.elasticlog.internal;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.time.Instant;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.annotations.Component;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+
+import dev.galasa.ManagerException;
+import dev.galasa.elasticlog.internal.properties.ElasticLogEndpoint;
+import dev.galasa.elasticlog.internal.properties.ElasticLogIndex;
+import dev.galasa.elasticlog.internal.properties.ElasticLogPropertiesSingleton;
+import dev.galasa.framework.spi.AbstractManager;
+import dev.galasa.framework.spi.IConfidentialTextService;
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IManager;
+import dev.galasa.framework.spi.ResourceUnavailableException;
+import dev.galasa.http.HttpClientException;
+import dev.galasa.http.HttpClientResponse;
+import dev.galasa.http.IHttpClient;
+import dev.galasa.http.spi.IHttpManagerSpi;
+
+/**
+ * ElasticLog Manager implementation
+ * 
+ * @author Richard Somers
+ */
+@Component(service = { IManager.class })
+public class ElasticLogManagerImpl extends AbstractManager {
+
+    private static final Log                    logger          = LogFactory.getLog(ElasticLogManagerImpl.class);
+    public final static String                  NAMESPACE       = "elastic";
+    
+    private IFramework							framework;
+    private IConfigurationPropertyStoreService  cps;
+    private IConfidentialTextService			ctf;
+
+    private IHttpManagerSpi        		        httpManager;
+    private IHttpClient							client;
+    
+    private Gson								gson;
+
+    private HashMap<String, Object>				runProperties   = new HashMap<String, Object>(); 
+
+    /**
+     * Initialies the ElasticLogManager, adding the requirement of the HttpManager
+     *  
+     * @param IFramework - the galasa framework
+     * @param List<IManager> - list of all the managers
+     * @param List<Imanager> - list of all the active managers
+     * @param Class<?> - the test class
+     * @throws ManagerException
+     */
+    @Override
+    public void initialise(@NotNull IFramework framework, @NotNull List<IManager> allManagers,
+            @NotNull List<IManager> activeManagers, @NotNull Class<?> testClass) throws ManagerException {
+        super.initialise(framework, allManagers, activeManagers, testClass);
+
+        youAreRequired(allManagers, activeManagers);
```

Comment:
Dont activate unless it is non-local run.   or a property exists to force it,   eg.  elasticlog.local.run.log=true